modules/core/src/lib/attribute.js

```diff
@@ -229,24 +229,31 @@ export default class LayerAttribute extends Attribute {
   // PRIVATE HELPER METHODS
 
   /* check user supplied values and apply fallback */
-  _normalizeValue(value, size = this.size, defaultValue = this.userData.defaultValue) {
+  _normalizeValue(
+    value,
+    size = this.size,
+    defaultValue = this.userData.defaultValue,
+    out = [],
```

Comment:
I'd be careful with having a default arguments here, it could be a perf killer. Can we remove them (at least the defaults for `defaultValue` and `out`)?

Reply:
Nit: A lot of args makes it a little hard to follow what is what when this function is called. Can we reduce? Does `size` need to be passed in?